trendmicrovisionone_connector.py

```diff
@@ -42,331 +37,154 @@ def __new__(cls, val1, val2=None):
-            return action_result.get_status()
-
-        # Add the response into the data section
-        action_result.add_data(response)
-
-        # return success
-        return action_result.set_status(phantom.APP_SUCCESS)
-
-    def lookup_type(self, param):
-
-        # Regex expression for validating IPv4
-        regex = (
-            "(([0-9]|[1-9][0-9]|1[0-9][0-9]|"
-            "2[0-4][0-9]|25[0-5])\\.){3}"
-            "([0-9]|[1-9][0-9]|1[0-9][0-9]|"
-            "2[0-4][0-9]|25[0-5])"
-        )
-
-        # Regex expression for validating IPv6
-        regex1 = "((([0-9a-fA-F]){1,4})\\:){7}" "([0-9a-fA-F]){1,4}"
-
-        # Regex expression for validating MAC
-        regex2 = "([0-9A-Fa-f]{2}[:-]){5}" "([0-9A-Fa-f]{2})"
-
-        p = re.compile(regex)
-        p1 = re.compile(regex1)
-        p2 = re.compile(regex2)
-
-        # Checking if it is a valid IPv4 address
-        if re.search(p, param):
-            return "ip"
-
-        # Checking if it is a valid IPv6 address
-        elif re.search(p1, param):
-            return "ipv6"
+        # Required Params
+        endpoint = param["ip_hostname_mac"]
+        query_op = param["query_op"]
 
-        # Checking if it is a valid MAC address
-        elif re.search(p2, param):
-            return "macaddr"
+        # Initialize pytmv1
+        client = self._get_client()
 
-        # Otherwise use hostname type
-        return "hostname"
-
-    def _handle_get_endpoint_info(self, param):
-        self.save_progress(
-            "In action handler for: {0}".format(self.get_action_identifier())
-        )
-
-        # Add an action result object to self (BaseConnector) to represent the action for this param
-        action_result = self.add_action_result(ActionResult(dict(param)))
+        # Choose QueryOp Enum based on user choice
+        if "or" in query_op:
+            query_op = pytmv1.QueryOp.OR
+        elif "and" in query_op:
+            query_op = pytmv1.QueryOp.AND
 
-        # Param setup
-        value = param["ip_hostname_mac"]
-        if "-" in value:
-            parts = value.split("-")
-            value = ":".join(parts)
-
-        # Unpack first item from a list of any depth
-        value = self.delistify(value)
-        field = self.lookup_type(value)
-        computer_id = self.get_computer_id(field, value)
-        body = {"computerId": computer_id}
-
-        # make rest call
-        ret_val, response = self._make_rest_call(
-            GET_ENDPOINT_INFO_ENDPOINT,
-            action_result,
-            method="post",
-            data=json.dumps(body),
-            params=None,
-            headers=self.header(),
-        )
+        new_endpoint_data: List[Any] = []
 
-        if phantom.is_fail(ret_val):
-            self.save_progress("Endpoint info lookup failed.")
+        # Make rest call
+        try:
+            client.consume_endpoint_data(
+                lambda endpoint_data: new_endpoint_data.append(endpoint_data.json()),
```

Comment:
Need to avoid calling BaseModel.json() because you're serializing and deserializing multiple times